### DISCUSSION TITLE
请教一个问题

### DISCUSSION THREAD

**Discussion by alex3236** [imfuxiao/Hamster#288] [Q&A] [answered]
最近接触了 [声笔系列码](https://github.com/sbxlm)，感觉很有意思，但官方没有做 IOS 支持。声笔的特殊性导致他们[修改了 librime](https://github.com/sbxlmdsl/librime) 以实现独有特性。

因此，我尝试修改 LibrimeKit 和 Hamster 以适配声笔系列码。我的问题如下：

因为我是第一次接触 IOS 开发，也没有 MacOS 环境，因此希望寻找一种其他方式构建 ipa 包。Hamster 似乎不能像其他一般 APP 那样直接将 xarchive 下的 Hamster.app 套一层 Payload 作为 ipa 使用？即使官方仓库 Action 中的 xarchive，这样打包之后会在第一次打开的方案部署阶段闪退。

**Comment by imfuxiao** [comment]
因为依赖项很多，尤其是对 librime 的依赖。需要先对 librime 做 arm 及 模拟器环境下的编译，编译为二进制依赖包供 hamster 使用。

像您所说的如果对 librime 的源码做了变更，那么同样在编译 librime 时做相应的变更。目前 1.x 版本使用 librime 的 release 1.8.5 版本。

具体看这个项目：https://github.com/imfuxiao/LibrimeKit

Hamster 2.x 版本（还在开发中）会将上面那个项目只作为 librime 层的编译项目。编译为供 hamster 使用的二进制依赖包，不在包含封装代码。

**Reply by alex3236** [reply]
我的问题其实是 没有XCode环境能不能打包出能用的ipa 在啥都不改的情况下（

**Reply by imfuxiao** [reply]
个人没有尝试过，理论上应该是可以的，你可以看看这个项目的 .github 下的ci配置yaml文件，是别的伙伴提交的PR，是可以编译成功的。因没有配置证书，所以最终没有打包成IPA。

**Reply by alex3236** [reply]
我现在用 Action 构建的包似乎没法用 签名安装都正常 但打开后到部署阶段就会闪退。同时在文件应用中显示的文件夹是无图标的“仓输入法”而不是AppStore下载版本有图标的“Hamster”。尚不清楚原因

**Reply by imfuxiao** [reply, marked as answer]
目前项目中的action确实有些问题，目前只是用来做提交PR代码是否能够编译成功，每次打包还是在我本地进行。

我猜测是跟个人开发证书有关，我本地编译的证书及包的签名都是自动进行的。没有使用手工配置证书。
